src/pool_type.zig

```diff
@@ -0,0 +1,162 @@
+const std = @import("std");
+const assert = std.debug.assert;
+
+pub fn find_first_unset(self: std.DynamicBitSetUnmanaged) ?u64 {
+    var offset: u64 = 0;
+    var mask = self.masks;
+    while (offset < self.bit_length) {
+        if (~mask[0] != 0) break;
+        mask += 1;
+        offset += @bitSizeOf(std.DynamicBitSetUnmanaged.MaskInt);
+    } else return null;
+    const first_unset = offset + @ctz(~mask[0]);
+    return if (first_unset < self.bit_length) first_unset else null;
+}
```

Comment:
Note that we also have `find_bit()` in `src/vsr/free_set.zig`, which is a generalization of this function. Maybe it belongs in stdx?

Reply:
FWIW, I'd go all the way and add `stdx/bit_set_dynamic.zig`, which wraps the underlying DynamicBitSet:

```zig
const BitSetDynamic = struct {
   inner: std.DynamicBitSetUnmanaged, 
}
```

* We in general want to warp std APIs, to reduce update churn
* I've historically found `BitSet`'s API awkward in terms of naming/signatures/orthogonality